src/rebound.c

```diff
@@ -572,6 +572,9 @@ int reb_check_exit(struct reb_simulation* const r, const double tmax, double* la
     }
     const double dtsign = copysign(1.,r->dt);   // Used to determine integration direction
     if (r->status>=0 || reb_error_message_waiting(r)){
+        if(r->status <= 0){
+            r->status = 1;
+        }
```

Comment:
I'm not sure I understand the consequences of this change.

Reply:
What was happening with the code snippet above is that r->status = -1, reb_error_message_waiting(r) would trigger because whfast reb_errored on not using Jacobi coordinates, so check_exit would just return with r->status = -1. Then whfast would return immediately on whfast_init on the reb_erorr without doing anything and it would loop. So I thought one solution (perhaps naive) is that if there is a reb_error_message_waiting, we could just set reb_status to the generic error = 1. Travis passes, but I might be missing some logic/edge cases...

Reply:
I wasn't sure if that could be the default behavior so that we don't have to treat and translate positive r->status values explicitly in python when sim.integrate returns and could just process_messages to raise whatever reb_errors occurred in C

Reply:
I see now that you didn't write that you didn't understand why it was there, but what the consequences would be...travis passes, but I agree it's a major change and quite unclear across all cases!

Reply:
Thanks. I think this is doing the right thing. Although I would probably write it like this:

```
if (reb_error_message_waiting(r)){
    r->status = REB_EXIT_ERROR;
}
if (r->status>=0){
        ...
```

for readability.

Reply:
Good idea! Thanks